src/reducers/locales.js

```diff
@@ -0,0 +1,53 @@
+import {addLocaleData} from 'react-intl';
+
+import {localeData} from 'scratch-l10n';
+import editorMessages from 'scratch-l10n/locales/editor-msgs';
+
+addLocaleData(localeData);
+
+const UPDATE_LOCALES = 'scratch-gui/locales/UPDATE_LOCALES';
+const SELECT_LOCALE = 'scratch-gui/locales/SELECT_LOCALE';
+
+const initialState = {
+    locale: 'en',
+    messages: editorMessages
```

Comment:
To avoid the need to look up messages by locale, while still allowing something to update the `messages`, maybe we should hold two things in state, `messagesByLocale` and `messages`. Where every time `locale` or `messagesByLocale` is updated, we set `messages` to `messagesByLocale[state.locale]`. Then connected components can just map `messages: state.messages` rather than needing to do the lookup and have knowledge about the internal workings of intl?

Reply:
That's to say, here maybe we can have
```js
initialState = {
    locale: 'en',
    messagesByLocale: editorMessages,
    messages: editorMessages['en']
}
```